File: app/src/main/java/com/github/mbarrben/moviedb/movies/MoviesFragment.java
package com.github.mbarrben.moviedb.movies;

import android.os.Bundle;
import android.view.View;
import butterknife.InjectView;
import com.github.mbarrben.moviedb.BaseFragment;
import com.github.mbarrben.moviedb.R;
import com.github.mbarrben.moviedb.domain.movies.MoviesPresenter;
import javax.inject.Inject;

public class MoviesFragment extends BaseFragment {

  @Inject MoviesPresenter presenter;
  @InjectView(R.id.movies_view) MoviesViewLayout moviesView;

  @Override
  protected int getFragmentLayout() {
    return R.layout.movies_fragment;
  }

  @Override
  public void onViewCreated(View view, Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    presenter.onViewCreated(moviesView);
  }

  @Override
  public void onDestroyView() {
    super.onDestroyView();
    presenter.onDestroyView();
  }
}

File: app/src/main/kotlin/com/github/mbarrben/moviedb/movies/view/platform/MoviesFragment.kt
package com.github.mbarrben.moviedb.movies.view.platform

import android.os.Bundle
import android.view.View
import com.github.mbarrben.moviedb.BaseFragment
import com.github.mbarrben.moviedb.R
import com.github.mbarrben.moviedb.domain.movies.MoviesPresenter
import com.github.mbarrben.moviedb.movies.di.MoviesListComponent
import kotlinx.android.synthetic.main.movies_fragment.moviesView
import javax.inject.Inject

class MoviesFragment : BaseFragment(R.layout.movies_fragment) {

  @Inject lateinit var presenter: MoviesPresenter

  override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    presenter.bind(moviesView)
  }

  override fun onDestroyView() {
    super.onDestroyView()
    presenter.unbind()
  }

  override fun inject() = getComponent(MoviesListComponent::class.java).inject(this)
}
